Crash when clicking on review notification

When testing this [PR](https://github.com/woocommerce/woocommerce-android/pull/1175), I got this crash when clicking on the notification for a product review.

```
2019-07-02 15:53:45.889 17416-17416/com.woocommerce.android E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.woocommerce.android, PID: 17416
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.woocommerce.android/com.woocommerce.android.ui.main.MainActivity}: java.lang.IllegalStateException: Fragment NotifsListFragment{8b68b20 id=0x7f0a00a5 NotifsListFragment} does not have a NavController set
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2913)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3048)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1808)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.IllegalStateException: Fragment NotifsListFragment{8b68b20 id=0x7f0a00a5 NotifsListFragment} does not have a NavController set
        at androidx.navigation.fragment.NavHostFragment.findNavController(NavHostFragment.java:119)
        at androidx.navigation.fragment.FragmentKt.findNavController(Fragment.kt:29)
        at com.woocommerce.android.ui.notifications.NotifsListFragment.openReviewDetail(NotifsListFragment.kt:319)
        at com.woocommerce.android.ui.main.MainActivity.showNotificationDetail(MainActivity.kt:557)
        at com.woocommerce.android.ui.main.MainActivity.initFragment(MainActivity.kt:514)
        at com.woocommerce.android.ui.main.MainActivity.onCreate(MainActivity.kt:137)
        at android.app.Activity.performCreate(Activity.java:7136)
        at android.app.Activity.performCreate(Activity.java:7127)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271)
```

### Steps to reproduce the crash:
- Enable `Do not keep activities`.
- Open the app login to a test site and close the app. Ensure that the app is cleared from the back stack.
- Write a product review from a test site.
- Once you receive a notification for the test site, click on it in the app.
- Notice that the app crashes.

@nbradbury I was able to reproduce this issue on my emulator (Pixel 2 running API 28):
```
         AndroidRuntime  D  Shutting down VM
                         E  FATAL EXCEPTION: main
                         E  Process: com.woocommerce.android, PID: 6446
                         E  java.lang.RuntimeException: Unable to start activity ComponentInfo{com.woocommerce.android/com.woocommerce.android.ui.main.MainActivity}: java.lang.IllegalStateException: Fragment NotifsListFr
                            agment{2e0cc3b id=0x7f0a00a5 NotifsListFragment} does not have a NavController set
                         E      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2913)
                         E      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3048)
                         E      at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
                         E      at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
                         E      at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
                         E      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1808)
                         E      at android.os.Handler.dispatchMessage(Handler.java:106)
                         E      at android.os.Looper.loop(Looper.java:193)
                         E      at android.app.ActivityThread.main(ActivityThread.java:6669)
                         E      at java.lang.reflect.Method.invoke(Native Method)
                         E      at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
                         E      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
                         E  Caused by: java.lang.IllegalStateException: Fragment NotifsListFragment{2e0cc3b id=0x7f0a00a5 NotifsListFragment} does not have a NavController set
                         E      at androidx.navigation.fragment.NavHostFragment.findNavController(NavHostFragment.java:119)
                         E      at androidx.navigation.fragment.FragmentKt.findNavController(Fragment.kt:29)
                         E      at com.woocommerce.android.ui.notifications.NotifsListFragment.openReviewDetail(NotifsListFragment.kt:319)
                         E      at com.woocommerce.android.ui.main.MainActivity.showNotificationDetail(MainActivity.kt:555)
                         E      at com.woocommerce.android.ui.main.MainActivity.initFragment(MainActivity.kt:514)
                         E      at com.woocommerce.android.ui.main.MainActivity.onCreate(MainActivity.kt:137)
                         E      at android.app.Activity.performCreate(Activity.java:7136)
                         E      at android.app.Activity.performCreate(Activity.java:7127)
                         E      at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271)
                         E      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2893)
                         E      ... 11 more
```

I followed these steps:
1. Download the [2.2-rc-1 beta release](https://github.com/woocommerce/woocommerce-android/releases/tag/2.2-rc-1)
2. Open emulator and uninstall WooCommerce. 
3. Drag and drop the apk downloaded in step 1 into the emulator to install. 
4. Log into the app with a test store (I used my testwooshop one which I'm pretty sure you have access to as well)
5. I didn't click on any of the tabs or anything in the app once logged in. Just minimized it and then force closed it from within the app overview.
6. Submit a review on my store and wait for the device to get the notification.
7. Click on the notification -- crash.

--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt
@@ -23,6 +23,7 @@ import com.woocommerce.android.extensions.FragmentScrollListener
 import com.woocommerce.android.extensions.WooNotificationType.NEW_ORDER
 import com.woocommerce.android.extensions.WooNotificationType.PRODUCT_REVIEW
 import com.woocommerce.android.extensions.active
+import com.woocommerce.android.extensions.getCommentId
 import com.woocommerce.android.extensions.getRemoteOrderId
 import com.woocommerce.android.extensions.getWooType
 import com.woocommerce.android.push.NotificationHandler
@@ -36,6 +37,7 @@ import com.woocommerce.android.ui.main.BottomNavigationPosition.DASHBOARD
 import com.woocommerce.android.ui.main.BottomNavigationPosition.NOTIFICATIONS
 import com.woocommerce.android.ui.main.BottomNavigationPosition.ORDERS
 import com.woocommerce.android.ui.notifications.NotifsListFragment
+import com.woocommerce.android.ui.notifications.ReviewDetailFragmentDirections
 import com.woocommerce.android.ui.orders.OrderDetailFragmentDirections
 import com.woocommerce.android.ui.orders.OrderListFragment
 import com.woocommerce.android.ui.prefs.AppSettingsActivity
@@ -53,6 +55,7 @@ import dagger.android.AndroidInjector
 import dagger.android.DispatchingAndroidInjector
 import dagger.android.support.HasSupportFragmentInjector
 import kotlinx.android.synthetic.main.activity_main.*
+import org.wordpress.android.fluxc.model.notification.NotificationModel
 import org.wordpress.android.fluxc.model.order.OrderIdentifier
 import org.wordpress.android.login.LoginAnalyticsListener
 import org.wordpress.android.login.LoginMode
@@ -541,7 +544,6 @@ class MainActivity : AppCompatActivity(),
         showBottomNav()
         bottomNavView.currentPosition = NOTIFICATIONS
 
-        val fragment = bottomNavView.getFragment(NOTIFICATIONS)
         val navPos = BottomNavigationPosition.NOTIFICATIONS.position
         bottom_nav.active(navPos)
 
@@ -554,7 +556,7 @@ class MainActivity : AppCompatActivity(),
                         }
                     }
                 }
-                PRODUCT_REVIEW -> (fragment as? NotifsListFragment)?.openReviewDetail(note)
+                PRODUCT_REVIEW -> showReviewDetail(note)
                 else -> { /* do nothing */ }
             }
         }
@@ -563,6 +565,16 @@ class MainActivity : AppCompatActivity(),
     override fun showProductDetail(remoteProductId: Long) {
         showBottomNav()
         val action = ProductDetailFragmentDirections.actionGlobalProductDetailFragment(remoteProductId)
+        navController.navigate(action)
+    }
+
+    override fun showReviewDetail(notification: NotificationModel, tempStatus: String?) {
+        showBottomNav()
+        val action = ReviewDetailFragmentDirections.actionGlobalReviewDetailFragment(
+                notification.remoteNoteId,
+                notification.getCommentId(),
+                tempStatus
+        )
         navController.navigate(action)
     }
 

--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainNavigationRouter.kt
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainNavigationRouter.kt
@@ -1,9 +1,12 @@
 package com.woocommerce.android.ui.main
 
+import org.wordpress.android.fluxc.model.notification.NotificationModel
+
 interface MainNavigationRouter {
     fun isAtNavigationRoot(): Boolean
     fun isChildFragmentShowing(): Boolean
 
     fun showOrderDetail(localSiteId: Int, remoteOrderId: Long, remoteNoteId: Long = 0, markComplete: Boolean = false)
     fun showProductDetail(remoteProductId: Long)
+    fun showReviewDetail(notification: NotificationModel, tempStatus: String? = null)
 }

--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/notifications/NotifsListContract.kt
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/notifications/NotifsListContract.kt
@@ -27,6 +27,7 @@ interface NotifsListContract {
         fun refreshFragmentState()
         fun showSkeleton(show: Boolean)
         fun showEmptyView(show: Boolean)
+        fun openOrderDetail(notification: NotificationModel)
         fun openReviewDetail(notification: NotificationModel)
         fun visuallyMarkNotificationsAsRead()
         fun showMarkAllNotificationsReadError()

--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/notifications/NotifsListFragment.kt
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/notifications/NotifsListFragment.kt
@@ -277,18 +277,8 @@ class NotifsListFragment : TopLevelFragment(),
                 }
             }
             NEW_ORDER -> {
-                notification.getRemoteOrderId()?.let {
-                    AnalyticsTracker.track(Stat.NOTIFICATION_OPEN, mapOf(
-                            AnalyticsTracker.KEY_TYPE to AnalyticsTracker.VALUE_ORDER,
-                            AnalyticsTracker.KEY_ALREADY_READ to notification.read))
-                    showOptionsMenu(false)
-                    (activity as? MainNavigationRouter)?.showOrderDetail(
-                            selectedSite.get().id,
-                            it,
-                            notification.remoteNoteId
-                    )
-                } ?: WooLog.e(NOTIFICATIONS, "New order notification is missing the order id!").also {
-                    showLoadNotificationDetailError()
+                if (!notifsRefreshLayout.isRefreshing) {
+                    openOrderDetail(notification)
                 }
             }
             UNKNOWN -> {
@@ -300,6 +290,22 @@ class NotifsListFragment : TopLevelFragment(),
         AppRatingDialog.incrementInteractions()
     }
 
+    override fun openOrderDetail(notification: NotificationModel) {
+        notification.getRemoteOrderId()?.let {
+            AnalyticsTracker.track(Stat.NOTIFICATION_OPEN, mapOf(
+                    AnalyticsTracker.KEY_TYPE to AnalyticsTracker.VALUE_ORDER,
+                    AnalyticsTracker.KEY_ALREADY_READ to notification.read))
+            showOptionsMenu(false)
+            (activity as? MainNavigationRouter)?.showOrderDetail(
+                    selectedSite.get().id,
+                    it,
+                    notification.remoteNoteId
+            )
+        } ?: WooLog.e(NOTIFICATIONS, "New order notification is missing the order id!").also {
+            showLoadNotificationDetailError()
+        }
+    }
+
     override fun openReviewDetail(notification: NotificationModel) {
         AnalyticsTracker.track(Stat.NOTIFICATION_OPEN, mapOf(
                 AnalyticsTracker.KEY_TYPE to AnalyticsTracker.VALUE_REVIEW,
@@ -308,8 +314,8 @@ class NotifsListFragment : TopLevelFragment(),
         // If the notification is pending moderation, override the status to display in the detail view.
         val isPendingModeration = pendingModerationRemoteNoteId?.let { it == notification.remoteNoteId } ?: false
         val tempStatus = if (isPendingModeration) pendingModerationNewStatus else null
-        (activity as? MainNavigationRouter)?.showReviewDetail(notification, tempStatus)
         showOptionsMenu(false)
+        (activity as? MainNavigationRouter)?.showReviewDetail(notification, tempStatus)
     }
 
     override fun scrollToTop() {

--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/notifications/NotifsListFragment.kt
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/notifications/NotifsListFragment.kt
@@ -10,7 +10,6 @@ import android.view.MenuItem
 import android.view.View
 import android.view.ViewGroup
 import androidx.core.content.ContextCompat
-import androidx.navigation.fragment.findNavController
 import androidx.recyclerview.widget.RecyclerView
 import com.google.android.material.snackbar.Snackbar
 import com.woocommerce.android.R
@@ -19,7 +18,6 @@ import com.woocommerce.android.analytics.AnalyticsTracker.Stat
 import com.woocommerce.android.extensions.WooNotificationType.NEW_ORDER
 import com.woocommerce.android.extensions.WooNotificationType.PRODUCT_REVIEW
 import com.woocommerce.android.extensions.WooNotificationType.UNKNOWN
-import com.woocommerce.android.extensions.getCommentId
 import com.woocommerce.android.extensions.getRemoteOrderId
 import com.woocommerce.android.extensions.getWooType
 import com.woocommerce.android.extensions.onScrollDown
@@ -310,13 +308,8 @@ class NotifsListFragment : TopLevelFragment(),
         // If the notification is pending moderation, override the status to display in the detail view.
         val isPendingModeration = pendingModerationRemoteNoteId?.let { it == notification.remoteNoteId } ?: false
         val tempStatus = if (isPendingModeration) pendingModerationNewStatus else null
-        val action = ReviewDetailFragmentDirections.actionGlobalReviewDetailFragment(
-                notification.remoteNoteId,
-                notification.getCommentId(),
-                tempStatus
-        )
+        (activity as? MainNavigationRouter)?.showReviewDetail(notification, tempStatus)
         showOptionsMenu(false)
-        findNavController().navigate(action)
     }
 
     override fun scrollToTop() {

--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/notifications/NotifsListFragment.kt
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/notifications/NotifsListFragment.kt
@@ -291,14 +291,14 @@ class NotifsListFragment : TopLevelFragment(),
     }
 
     override fun openOrderDetail(notification: NotificationModel) {
-        notification.getRemoteOrderId()?.let {
+        notification.getRemoteOrderId()?.let { remoteOrderId ->
             AnalyticsTracker.track(Stat.NOTIFICATION_OPEN, mapOf(
                     AnalyticsTracker.KEY_TYPE to AnalyticsTracker.VALUE_ORDER,
                     AnalyticsTracker.KEY_ALREADY_READ to notification.read))
             showOptionsMenu(false)
             (activity as? MainNavigationRouter)?.showOrderDetail(
                     selectedSite.get().id,
-                    it,
+                    remoteOrderId,
                     notification.remoteNoteId
             )
         } ?: WooLog.e(NOTIFICATIONS, "New order notification is missing the order id!").also {

